pyteal/ast/router.py

```diff
@@ -96,6 +83,13 @@ class MethodConfig:
     update_application: CallConfig = field(kw_only=True, default=CallConfig.NEVER)
     delete_application: CallConfig = field(kw_only=True, default=CallConfig.NEVER)
 
+    def __post_init__(self):
+        if self.clear_state != CallConfig.NEVER:
+            raise TealInputError(
+                "Attempt to construct clear state program from MethodConfig, "
+                "use Router top level argument `clear_state` instead"
+            )
+
```

Comment:
Question: this function will error if one sets `clear_state` to anything other than default. Are we good here, or any more steps needed to totally remove the keyword arg `clear_state`?

Reply:
This seems sufficient to me

Reply:
We might want to have a link to a URL that explains in more detail. But that can also be added in the followup pr/release.

Reply:
I update the doc a little bit and referred to doc in [030eaf4](https://github.com/algorand/pyteal/pull/636/commits/030eaf4c3e3743587abdbd85a4b933bc8510b72a), keep looking into what should be done for csp refactoring.